src/backlight/datasource/__init__.py

```diff
@@ -18,13 +22,15 @@ def load_marketdata(symbol, start_dt, end_dt, url):
     return from_dataframe(symbol, df)
 
 
-def from_dataframe(df, symbol, col_mapping=None):
+def from_dataframe(
+    df: pd.DataFrame, symbol: str, col_mapping: dict = None
```

Comment:
```suggestion
    df: pd.DataFrame, symbol: str, col_mapping: Optional[dict] = None
```
See [`typing.Optional`](https://docs.python.org/3/library/typing.html#typing.Optional) for why, but basically `None` is not a `dict`.
You can add it to all subsequent values defaulting to `None`